packages/bruno-js/src/bruno-request.js

```diff
@@ -56,11 +56,31 @@ class BrunoRequest {
     this.req.headers[name] = value;
   }
 
-  getBody() {
+  /**
+   * Get the body of the request
+   * 
+   * @param {*} options 
+   * @param {boolean} options.raw - If true, return the raw body without parsing it
+   * @returns 
+   */
+  getBody(options = {}) {
+    let headers = this.req.headers;
+    const contentType = headers?.['Content-Type'] || headers?.['content-type'] || '';
+    const hasJSONContentType = contentType.includes('json');
+
+    if (hasJSONContentType && !options.raw) {
+      return JSON.parse(this.req.data);
+    }
+
     return this.req.data;
   }
 
   setBody(data) {
+    if (typeof data === 'object') {
```

Comment:
This should probably also check if the Object is a "plain" object with `data.constructor === Object`. Setting a FormData object as the body does not work now, but worked previously.

Reply:
This makes sense. Will have this updated.

Reply:
So I have updated the logic to also check if content type is json (based on headers).
This should cover the use case of people setting body as FormData object.

If someone has set the Content-Type as json in the header (or it gets auto added if they have chosen JSON as body in UI), and then they try to pass a non-plain object while invoking `setBody()` - they are essentially shooting themselves in the foot and we don't need to gracefully handle that.

The reason why are supporting the `raw` option is to support use cases where people want to still have the content type as json, but want to pass the json strings themselves instead of passing the json object

@Its-treason lmk if you differ or have further suggestions

```javascript
setBody(data, options = {}) {
    if (options.raw) {
      this.req.data = data;
      this.body = data;
      return;
    }

    const isJson = hasJSONContentType(this.req.headers);
    if (isJson && this.__isObject(data)) {
      this.body = data;
      this.req.data = this.__safeStringifyJSON(data);
      return;
    }

    this.req.data = data;
    this.body = data;
  }
```